lib/Doctrine/Common/Annotations/DocParser.php

```diff
@@ -615,8 +616,15 @@ class_exists(NamedArgumentConstructor::class);
                 foreach ($constructor->getParameters() as $parameter) {
                     $metadata['constructor_args'][$parameter->getName()] = [
                         'position' => $parameter->getPosition(),
-                        'default' => $parameter->isOptional() ? $parameter->getDefaultValue() : null,
+                        'is_variadic' => $parameter->isVariadic(),
```

Comment:
Do we need to remember the variadic parameter at all? Wouldn't it be enough to not add it to `$metadata['constructor_args']`?

Reply:
I think we can ignore the `is_variadic` parameter and check the `default` key only. The key will be missing for a variadic

Reply:
Done.

Reply:
Sorry for the nit-picking here. But this generates the edge-case of a potentially undefined `default` key that we didn't have previously.

Again, if we moved the `$parameter->isVariadic()` check to the very beginning of the `foreach` and thus skipped adding the variadic parameter to the `$metadata['constructor_args']` array, we wouldn't need to skip it when iterating over that array later on. Or am I missing something?

Reply:
> Sorry for the nit-picking here

Don't worry. That's OK.

> But this generates the edge-case of a potentially undefined default key that we didn't have previously.

This is an unreachable case for existing code as it only applies to variadic parameters.
An it is correct because a variadic param can't have a default value (even null).

> thus skipped adding the variadic parameter to the $metadata['constructor_args'] array

If we don't want to cutomize a new error for variadics only then yes, we can just skip that parameter on parsing